### ISSUE TITLE
C8 Error During Navigation

### ISSUE THREAD

**Issue by ghost** [ClusterM/hakchi2#108]
First, great work guys! Love how easy Hakchi2 (2.11) is to work with!

I modded my NESCM with around 700 NES games, none that Hakchi2 warned me might not play correctly, kept the menu to display 30 games per screen, did not add a button combination to get back to the menu screen, kept the epilepsy filter on, and included box art for each game I added. The following day I was browsing the game list, and when I pressed "A" on a "Back" folder, the screen darkened and froze. Not totally black, because I saw the Nintendo name in red at the lower left side of the screen. After a few seconds, it kicked back a C8 error. I shut off the system, and when I turned it back on I got a C2 error (which reset my settings).

Once the system was back up, I browsed through the game lists. The games and save states appeared to be intact. Getting the C8 got me worried about damaging my system, though (they still aren't easy to come by), so I uninstalled Hakchi2 from my NESCM for now.

Anyone know what may have caused the C8 error? Has this happened to anyone else? If it occurred while attempting to access a game, I would have figured it was a problem with that specific game and removed the one game from the system, but the error occurred while navigating through the folders.

Would the system be more stable if I have less games? Any thoughts on this is greatly appreciated.

Thank you for your time.

**Comment by 1lokolo1** [comment]
I have some of this issue, it wasn't a great issue. Only appears once, and if you don't add more games it doesn't happen again. In last build of 2.12 I don't have this issue at the moment so maybe it's solved. 

Rgs.

**Comment by maskedmaniac** [comment]
Thanks. I didn't realize 2.12 was already out. I'll give it a try. Thanks.

**Comment by 1lokolo1** [comment]
I'm sorry. The last build is need to be compile. Isn't a 2.12 release.

Rgs.

**Comment by maskedmaniac** [comment]
No problem. It sounds like 2.12 will be released soon. I'll wait for it to come out and give it a try.

Thanks!

**Comment by bfluryx1234** [comment]
The C8 followed by the C2 seems to be when the system hard crashes in my experience.  I had it happen once when trying to create a save state, but moved on and didn't see it happen again.  

The C8 error itself is due to the maximum number of games/save states that can be shown on the screen at once.  Recommendation is to limit folders to 30 as you have done.

**Comment by maskedmaniac** [comment]
So I upgraded to 2.13, but unfortunately ran into the C8/C2 error again tonight trying to access a save state.

I kept the folder limit to 30 per screen (with box art), and on the screen the C8/C2 error occurred, there were only three save states total out of all the 30 games displayed.

As much as I like the idea of having a modded NES, running into this error yet again is really making me nervous that I'm corrupting my system in the long run.

[Added...]

Spitballing here (I am definitely not an expert at all this), but while counting the games displayed on the screen the error occurred, I noticed 30 are shown, and an additional "back" folder is at the end, so technically there are 31 folders per screen. If the NES Classic's shell is designed to display a total of 30 games per screen (as explained in Hakchi2's FAQs), would it be more stable to show 29 games per screen with one "back" folder?

**Comment by ClusterM** [comment]
Do you still have this problem? Can you describe which actions you did? How long you played?

**Comment by ClusterM** [comment]
Seems like it's the same as #198

**Comment by maskedmaniac** [comment]
ClusterM, I experienced the error on versions 2.11 and 2.13. I've reverted back to factory, so I have yet to try to duplicate the the error in 2.14.

Both occasions were while I was browsing through the menu of NES games. Basically I'd browse through a sub folder, play one or two for 15 min or so, then move on to the next sub folder of games. The error occurred after 45-60 minutes - one occurred when I attempted to acces a "back" folder to get me back to the main list of folders, and the other time was while trying to access a save point for one of the games.

Hope this helps track down the culprit.
